src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,172 @@
+            Node<T> newNode = new Node<>(list.get(i));
+            tail = newNode;
+            if (temporaryItem == null) {
+                head = newNode;
+            } else {
+                temporaryItem.next = newNode;
+                newNode.prev = temporaryItem;
+            }
+            size++;
+        }
     }
 
     @Override
     public T get(int index) {
+        checkIndex(index);
+        Node<T> temporaryItem = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                return temporaryItem.item;
+            }
+            temporaryItem = temporaryItem.next;
+        }
         return null;
     }
 
     @Override
     public T set(T value, int index) {
+        checkIndex(index);
+        Node<T> temporaryItem = head;
+        Node<T> newNode = new Node<>(value);
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                T element = temporaryItem.item;
+                temporaryItem.item = newNode.item;
+                return element;
+            }
+            temporaryItem = temporaryItem.next;
+        }
         return null;
     }
 
     @Override
     public T remove(int index) {
+        checkIndex(index);
+        Node<T> temporaryItem = head;
+        for (int i = 0; i <= index; i++) {
+            if (i == index) {
+                T element = temporaryItem.item;
+                if (size == 1) {
+                    head = null;
+                    tail = null;
+                } else if (temporaryItem == head) {
+                    temporaryItem.next.prev = null;
+                    head = temporaryItem.next;
+                } else if (temporaryItem == tail) {
+                    temporaryItem.prev.next = null;
+                    tail = temporaryItem.prev;
+                } else {
+                    temporaryItem.prev.next = temporaryItem.next;
+                    temporaryItem.next.prev = temporaryItem.prev;
+                }
+                size--;
+                return element;
+            }
+            temporaryItem = temporaryItem.next;
+        }
         return null;
     }
 
     @Override
     public boolean remove(T object) {
+        int index = index(object);
+        if (index != -1) {
+            remove(index);
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private int index(T element) {
+        int index = -1;
+        Node<T> temporaryItem = head;
+        for (int i = 0; i < size; i++) {
+            if (temporaryItem.item == element
+                    || temporaryItem.item != null && temporaryItem.item.equals(element)) {
+                return i;
+            }
+            temporaryItem = temporaryItem.next;
+        }
+        return index;
```

Comment:
The variable `temporaryItem` could be named more informatively, for example, `currentNode`. Also, you could use a method like `findNodeByIndex` to avoid code duplication.